File: teledonJFX/src/main/java/com/example/teledonjfx/service/DonationService.java
package com.example.teledonjfx.service;

import com.example.teledonjfx.model.CharitableCase;
import com.example.teledonjfx.model.Donation;
import com.example.teledonjfx.model.Donor;
import com.example.teledonjfx.repository.*;
import com.example.teledonjfx.utils.Observable;
import com.example.teledonjfx.utils.Observer;
import com.example.teledonjfx.utils.events.DonationEvent;

import java.util.ArrayList;
import java.util.List;

public class DonationService implements Observable<DonationEvent> {
    private IDonationRepository donationRepository;
    private IDonorRepository donorRepository;
    private ICharitableCaseRepository charitableCaseRepository;

    public DonationService(DonationRepository donationRepository, DonorRepository donorRepository, CharitableCaseRepository charitableCaseRepository) {
        this.donationRepository = donationRepository;
        this.donorRepository = donorRepository;
        this.charitableCaseRepository = charitableCaseRepository;
    }

    private List<Observer<DonationEvent>> observerList = new ArrayList<>();

    @Override
    public void addObserver(Observer<DonationEvent> e) {
        observerList.add(e);
    }

    @Override
    public void removeObserver(Observer<DonationEvent> e) {
        observerList.remove(e);

    }

    @Override
    public void notifyObservers(DonationEvent t) {
        observerList.forEach(x -> x.update(t));

    }
    public void saveDonation(CharitableCase charitableCase, Donor donor,Integer amount){
        Donation newDonation = new Donation(donor,charitableCase,amount);
        donationRepository.save(newDonation);
    }
}

File: teledonDesktop/teledonDesktop/services/DonationService.cs
using System;
using System.Collections.Generic;
using teledonCS.model;
using teledonCS.repository;
using teledonCS.utils.observer;
using teledonCS.utils.observer.events;

namespace teledonDesktop.services
{
    public class DonationService:Observable<DonationEvent>
    {
        public IDonationRepository DonationRepository;
        public IDonorRepository DonorRepository;
        public ICharitableChaseRepository charitableChaseRepository;
        public List<Observer<DonationEvent>> observers = new List<Observer<DonationEvent>>();

        public DonationService(IDonationRepository donationRepository, IDonorRepository donorRepository, ICharitableChaseRepository charitableChaseRepository)
        {
            DonationRepository = donationRepository;
            DonorRepository = donorRepository;
            this.charitableChaseRepository = charitableChaseRepository;
        }

        public void addObserver(Observer<DonationEvent> e)
        {
            observers.Add(e);
        }

        public void removeObserver(Observer<DonationEvent> e)
        {
            observers.Remove(e);
        }

        public void notifyObserver(DonationEvent t)
        {
            foreach (var observer in observers)
            {
                observer.update(t);
            }
        }

        public void saveDonation(CharitableCase charitableCase, Donor donor, int amount)
        {
           
            int oldAmount = charitableCase.amountRaised;
            int newAmount = oldAmount + amount;
            CharitableCase charitableCaseUpdated = new CharitableCase(charitableCase.name, newAmount);
            charitableCaseUpdated.id = charitableCase.id;
            Donation newDonation = new Donation(donor, charitableCase, amount);
            DonationRepository.save(newDonation);
            
            charitableChaseRepository.update(charitableCaseUpdated); 
            notifyObserver(new DonationEvent(ChangeEventType.ADD,charitableCase));

            
          

        }
    }
}